src/components/IComponent.hpp

```diff
@@ -27,20 +27,21 @@ namespace indie
             MUSIC,
             SOUND,
             EVT_LISTENER,
+            BONUS,
+            PLAYER,
             RECT,
             CAMERA,
             SPHERE,
             CUBE,
             GRID,
-            PLAYER,
             ANIMATION,
             BOMB,
             TIMER,
             DESTRUCTIBLE,
             AI,
             TYPE_NB
         };
-        virtual Type getType() const = 0;
+        virtual const Type &getType() const = 0;
```

Comment:
Why add reference of an enum ?

Reply:
To avoid a copy, but maybe it's overkill

Reply:
why not? this value is not gonna be modified, so better use it as const

Reply:
Type is an enum so a number a reference is fatter